src/Sarif/PropertiesDictionary.cs

```diff
@@ -37,11 +37,6 @@ public PropertiesDictionary() : this(null)
         {
         }
 
-        protected PropertiesDictionary(SerializationInfo info, StreamingContext context)
```

Comment:
>PropertiesDictionary [](http://example.com/codeflow?start=18&length=20)

Why did you remove this constructor? Preivously in .NET this ctor was required for all [Serializable] things, has that changed?

Reply:
the .net core concurrentdictionary is not binary serializable, so I removed it, 

question, do we even need it, it is not for json and xml serialization, it is only for BinaryFormatter serialization, in the solution I don't see code use BinaryFormatter

Reply:
Right, so if it's used anywhere it's in unit tests, I'd guess.

If you intend for this dictionary *not* to be serializable, you should remove the [Serializable] attribute. I think that's ok to do.

Reply:
That's a good question and in future, definitely try to raise this before making this kind of change. Remember we are in an SDK, so it is possible that someone is using BinaryFormatter with this public type (even though we do not as you point out).

I think probably it is unlikely we're breaking someone out there. So this change is fine. *However* we need to very clearly mark this breaking change in behavior in the release notes.

Imagine someone using BInaryFormatter with this type, using an older SDK. Now they update to the new SDK and everything is broken. Ideally, your release notes would also mention the symptom these users would receive. Make sense? IOW, try to use BinaryFormatter again PropertiesDictionary (once you remove the [Serializable] attribute as well. See what exception is raised. Make sure the release notes reference this exception in case users experience it.

By doing this, we leave an easily followed trail of clues that tell developers what happened. That's the least we can do, given that we're introducing a breaking change. :)

Reply:
thanks for the great inputs, 
I have remove the [Serializable] attribute and ran it, the error is 
SerializationException: Type PropertiesDictionary is not marked as serializable
I have added it both in the PR desc and also the release history md file.

Reply:
@michaelcfanning 

additional question, do you want me to create a work item to check and remove all [Serializable] in the solution,
( I have not dig if any of them is used somewhere yet, just to bring up the discussion )

[https://docs.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide](https://docs.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide)

[https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2300](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2300)